### PR TITLE
1932/ga updates

### PR DESCRIPTION
**Pull request by seanmalbert**
## Issue Overview

This PR addresses #1932 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This adds a way to push events to the GTM data layer and adds it to many of the pages based off of the info here https://airtable.com/shrpTlCXtMyA1KcBe/tblKovUImJGAwtBWw. The general page view event was added to all pages but the digital application, since that seems like it should have its own set of parameters. Listing list and detail events were also added.

## How Can This Be Tested/Reviewed?

You can test this by going to different pages on the public site and in the browser's console, look at `dataLayer`, and you should see the events in there. 
Screen Shot 2022-01-10 at 7 21 20 PM

Describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/lib/constants.ts [hunk 1]
```diff
@@ -1 +1,6 @@
 export const DATE_FORMAT = "MMMM D, YYYY"
+
+export enum USER_STATUS {
```
Comment (emilyjablonski) [inline sites/public/lib/constants.ts:3]: nit: our other enums are written with camel case

File: sites/public/pages/listing/[id]/[slug].tsx [hunk 2]
```diff
@@ -15,11 +17,35 @@ interface ListingProps {
...
+    if (!listing.id) return
+    pushGtmEvent<ListingDetail>({
+      event: "pageView",
+      pageTitle: pageTitle,
+      status: profile ? USER_STATUS.LoggedIn : USER_STATUS.NotLoggedIn,
+      listingStartDate: listing.applicationOpenDate,
+      listingStatus: listing.status,
+      listingID: listing.id,
+      applicationDueDate: listing.applicationDueDate,
+      paperApplication: listing.paperApplication,
```
Comment (emilyjablonski) [inline sites/public/pages/listing/[id]/[slug].tsx:33]: Just curious why this one is on here?

File: sites/public/pages/listings.tsx [hunk 3]
```diff
@@ -37,10 +40,21 @@ const closedListings = (listings) => {
...
 export default function ListingsPage(props: ListingsProps) {
+  const { profile } = useContext(AuthContext)
   const pageTitle = `${t("pageTitle.rent")} - ${t("nav.siteTitle")}`
   const metaDescription = t("pageDescription.welcome", { regionName: t("region.name") })
   const metaImage = "" // TODO: replace with hero image
 
+  useEffect(() => {
+    pushGtmEvent<ListingList>({
+      event: "pageView",
+      pageTitle: pageTitle,
```
Comment (emilyjablonski) [inline sites/public/pages/listings.tsx:51]: Not sure if it matters but we do translate page titles - so if we wanted to look up listings hits by page title we would need to look up all the language strings

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Needs a small change to build, and otherwise a couple non-blocking questions

File: sites/public/pages/listing/[id]/[slug].tsx (hunk 2 above)
Comment (seanmalbert) [reply sites/public/pages/listing/[id]/[slug].tsx:33]: PaperApplication? Lori, our GA expert, thinks this is useful information to track.

File: sites/public/pages/listings.tsx (hunk 3 above)
Comment (seanmalbert) [reply sites/public/pages/listings.tsx:51]: This is a good question and I asked Lori for her preference, so waiting to hear back on that. I think it'd be useful to track it.

File: sites/public/pages/account/edit.tsx [hunk 4]
```diff
@@ -39,6 +41,16 @@ const Edit = () => {
   const password = useRef({})
   password.current = watch("password", "")
 
+  useEffect(() => {
+    if (profile) {
```
[the remark this answers is not in the document]
Comment (seanmalbert) [reply sites/public/pages/account/edit.tsx:45]: Good catch, I updated to remove the ternary.

File: sites/public/lib/constants.ts (hunk 1 above)
Comment (seanmalbert) [reply sites/public/lib/constants.ts]: Updated to camel case.

**Review by emilyjablonski — APPROVED** [review]

**Pull request merged**